File: coding-practice/50 Coding Challenges part 1/Python/problem19.py
# create a function that will return an array of the first nPrime numbers greater than a number startAt
import math
def is_prime(x):
    if x < 2:
        return False
    if x == 2:
        return True
    for i in range(2, int(math.sqrt(x) + 1)):
        if x%i == 0:
            return False
    return True

def array_prime(n_prime, start_at):
    n = 0
    i = (start_at + 1)
    array = []

    while (n < n_prime):
        if is_prime(i):
            array.append(i) if (is_prime(i) == True and i > start_at) else None
            n += 1
        i += 1
    return array

array = array_prime(6, 11)
print(array)

File: coding-practice/50 Coding Challenges part 1/C#/problem19.cs
// coding challenge 19 : Create a function that will return in an array the first "nPrimes" prime numbers greater than a number "startAt"
using System;
class problem19
{   
    static bool is_prime(int number){
        if(number<2){
            return false;
        }

        else if(number==2){
            return true;
        }

        int max_div = (int)Math.Sqrt(number);
        for(int i=0; i<=max_div; i++){
            if(number%2 == 0){
                return false;
            }
        }

        return true;
    }

    static int[] array_prime(int nPrimes, int startAt){
        int[] output = new int[nPrimes];
        int j = 0;
        for(int i=0; i<nPrimes;){
            if((is_prime(j)==true) && (j>startAt)){
                output[i] = j;
                i++;
            }
            j++;
        }
        return output;
    }
    static void Main(string[] args){
        int nPrimes = 5;
        int startAt = 5;
        int[] array = new int[nPrimes];
        array = array_prime(nPrimes, startAt);
        Console.Write("The first "+nPrimes+" prime numbers after "+startAt+" are: ["+string.Join(", ",array)+"]");

    }
}